### ISSUE TITLE
Why it builds a new puma.config instead of using config/puma/production.rb? (question)

### ISSUE THREAD

**Issue by vizcay** [seuros/capistrano-puma#244]
After reading https://github.com/puma/puma#configuration-file it looks like the the "standard way" to configure the puma server for production is setting up "config/puma/production.rb". But this gem creates it's own configuration file "on the fly", that is deployed with "rake puma:config" by using the settings in the cap file.

Why diffier from the standard way? capistrano-puma was created before puma established the standard method for configuratin?

**Comment by seuros** [comment]
Yes.
You can set :puma_conf to use the file you want.

**Comment by vizcay** [comment]
IMHO working with the standard puma configuration file should be the "standard way".. it allows you more flexibility than recreating it from scratch with capistrano variables and uploading it manually with ```rake puma:conf```.

By ```set :puma_conf, "#{current_path}/config/puma/production.rb"``` into config/deploy.rb allows you to configure everithing there and has the advantage that going into your app dir in the VPS and doing ```RAILS_ENV=production bundle exec puma``` just works (as it should).

How do you feel about updating the README to reflect this? I can submit a PR maybe.

**Comment by arunraj88** [comment]
On setting `:puma_conf` I am getting an error saying `No such file or directory (RuntimeError)`. On checking, one thing I noticed was that the `current_path` was not available at the time of running `puma:check` task.  I think `puma:check` task was run after `deploy:check` and before fetching the code.  At the time of running `puma:check`, releases folder was empty.  Can you please look into this?

**Comment by seuros** [comment]
@arunraj88 . use `release_path`

**Comment by arunraj88** [comment]
@seuros checked with both `release_path` and `current_path`. Both return the same path(<deploy_to>/current) 

According to the `cap` [deploy flow](https://capistranorb.com/documentation/getting-started/flow/),  `git:create_release` was run only after `deploy:check`. So I think, the new release path will not be available at the time of running the `puma:check` task.

**Comment by Stex** [comment]
@seuros I'm not sure if this ticket should be closed as I don't think the actual problem has been addressed. Maybe it should become its own ticket though as the main problem is not that the gem *can* generate a shared file, but rather that it makes it very difficult to *not* use one.

I am in the same situation as the OP: When setting `puma_conf` to a file within my repository, both the initial deployment and `deploy:check` will fail as the file simply doesn't exist yet on the server without the git checkout:

```ruby
set :puma_conf, -> { "#{release_path}/config/puma/production.rb" }
```

I really don't like having to have a shared puma config on each server and would love to use the repository version, but the only way I found to get this done is with this *really ugly* solution for the first deployment:

```ruby
namespace :puma do
  task :setup_dummy_conf do
    on roles(:all) do
      execute 'mkdir -p /tmp/current/config/puma'
      execute "touch /tmp/current/config/puma/#{fetch(:rails_env)}.rb"
      execute "ln -s /tmp/current #{current_path}"
    end
  end

  before :check, :setup_dummy_conf
end
```

This way, `capistrano-puma` finds the file and is happy, but that shouln't be necessary.

> use release_path

As mentioned above, `release_path` will only differ from `current_path` after `deploy:starting` is through.
